CCS/Guadaloop_lib/guadaloop/lib/i2c/i2c_read_write.c

```diff
@@ -0,0 +1,246 @@
+        while((i2cSettings->i2cPort->MCS & I2C_MCS_ERROR) >> 1 !=0){
+
+        if(i2cSettings->i2cPort->MCS & (I2C_MCS_DATACK | I2C_MCS_ADRACK | I2C_MCS_ERROR )){
+
+                    i2cSettings->i2cPort->MCS = I2C_MCS_STOP;
+                    return 0;
+
+            }
+
+        }
+        readBuffer[i] = i2cSettings->i2cPort->MDR;
+        // i2cSettings->i2cPort->MCS &= 0xE9; // xxx01001
+        i2cSettings->i2cPort->MCS = I2C_MCS_RUN;
+    }
+
+    //i2cSettings->i2cPort->MCS &= 0xE5; //xxx00101
+    i2cSettings->i2cPort->MCS = I2C_MCS_STOP;
+
+
+    return 1;
+
+}
+
+uint8_t I2C_write_register(Transaction_t *i2cTransaction, I2C_Settings_t *i2cSettings, uint8_t *writeBuffer){
+    /*
+     * To write a register via I2C
+     * 1. Write slave addr to I2CMSA
+     * 2. Write register addr to I2CMDR
+     * 3. Write data to I2CMDR
+     *
+     * */
+
+     i2cSettings->i2cPort->MSA |= i2cTransaction->slaveAddress << 1;
+     i2cSettings->i2cPort->MSA &= ~I2C_MSA_SA_S;
+     //write data to I2CMDR
+     i2cSettings->i2cPort->MDR = i2cTransaction->regAddress;
+    // i2cSettings->i2cPort->MCS |= 0xEF;
+     i2cSettings->i2cPort->MCS = I2C_MCS_START|I2C_MCS_RUN;
+
+     while((i2cSettings->i2cPort->MCS) & I2C_MCS_BUSY != 0){}; //busy bit
+     while((i2cSettings->i2cPort->MCS & I2C_MCS_ERROR) >> 1 != 0){ //error bit
+
+         if(i2cSettings->i2cPort->MCS & (I2C_MCS_DATACK | I2C_MCS_ADRACK | I2C_MCS_ERROR )){
+
+             i2cSettings->i2cPort->MCS = I2C_MCS_STOP;
+             return 0;
+
+         }
+
+     }
+
+     uint8_t i;
+     for(i=0; i<i2cTransaction->writeCount; i++){
+         i2cSettings->i2cPort->MDR = writeBuffer[i];
+         while((i2cSettings->i2cPort->MCS) & I2C_MCS_BUSY != 0){};
+         if(i2cSettings->i2cPort->MCS & (I2C_MCS_DATACK | I2C_MCS_ADRACK | I2C_MCS_ERROR)){
+             i2cSettings->i2cPort->MCS = I2C_MCS_STOP;
+             return 0;
+         }
+         i2cSettings->i2cPort->MCS = I2C_MCS_RUN;
+     }
+     i2cSettings->i2cPort->MCS = I2C_MCS_STOP;
+
+     return 1;
+}
+
+
+void init_Transaction(Transaction_t *i2cTransaction){
+
+    i2cTransaction->writeCount = 0;
+    i2cTransaction->readCount = 0;
+
+}
+
+void init_Settings(I2C_Settings_t *i2cSettings){
+
+    switch(i2cSettings->i2cModule){
+
+    case I2CModule_0:
+           i2cSettings->i2cPort = I2C0;
+           break;
+    case I2CModule_1:
+           i2cSettings->i2cPort = I2C1;
+           break;
+    case I2CModule_2:
+           i2cSettings->i2cPort = I2C2;
+           break;
+    case I2CModule_3:
+           i2cSettings->i2cPort = I2C3;
+           break;
+    case I2CModule_4:
+           i2cSettings->i2cPort = I2C4;
+           break;
+    case I2CModule_5:
+           i2cSettings->i2cPort = I2C5;
+           break;
+    case I2CModule_6:
+           i2cSettings->i2cPort = I2C6;
+           break;
+    case I2CModule_7:
+           i2cSettings->i2cPort = I2C7;
+           break;
+
```

Comment:
bit neatpicky, but remove all the empty lines. makes code look cleaner and easier to read.